### PR TITLE
Track Operations added to Production Order

### LINKED ISSUE #3521: time log issue
Is there ant way to disable time log feature .. i tried disabling workstation but it still ask for time log

just like in V4 where only operational cost is mentioned and time log is disabled

### LINKED ISSUE #3581: Server error while making time log
I have a BOM that is a single operation on an item. The operation has a time associated but no current workstation.

I transferred material for manufacture and when trying to update finished goods I get the message to update it via "time logs". I click on the "pending" operation and when I click on "Make time log" I get the following error:

Traceback (innermost last):
File "/home/inpulse/frappe-bench/apps/frappe/frappe/app.py", line 51, in application
response = frappe.handler.handle()
File "/home/inpulse/frappe-bench/apps/frappe/frappe/handler.py", line 69, in handle
execute_cmd(cmd)
File "/home/inpulse/frappe-bench/apps/frappe/frappe/handler.py", line 92, in execute_cmd
ret = frappe.call(method, **frappe.form_dict)
File "/home/inpulse/frappe-bench/apps/frappe/frappe/__init__.py", line 751, in call
return fn(*args, **newargs)
TypeError: make_time_log() takes at least 4 arguments (4 given)

I'm stuck with this production order because it is already associated with the material transfer.
Thank you for any help!

### PR DESCRIPTION
**Pull request by neilLasrado**
Fixed #3581 
Fixed #3521

### CODE REVIEW

File: erpnext/manufacturing/doctype/production_order/production_order.py [hunk 1]
```diff
@@ -327,6 +329,12 @@ def validate_production_item(self):
 		
 		if frappe.db.get_value("Item", self.production_item, "has_variants"):
 			frappe.throw(_("Production Order cannot be raised against a Item Template"))
+	
+	def track_operation(self):
+		if self.track_operations:
+			self.set_production_order_operations()
+		else:
+			self.set('operations', [])
```
Comment (nabinhait) [inline erpnext/manufacturing/doctype/production_order/production_order.py]: set operations blank in client side only, ajax not required for this

File: erpnext/manufacturing/doctype/production_order/production_order.py [hunk 2]
```diff
@@ -391,7 +399,7 @@ def get_events(start, end, filters=None):
 	return data
 
 @frappe.whitelist()
-def make_time_log(name, operation, from_time, to_time, qty=None,  project=None, workstation=None, operation_id=None):
+def make_time_log(name, operation, from_time=now(), to_time=now(), qty=None,  project=None, workstation=None, operation_id=None):
```
Comment (nabinhait) [inline erpnext/manufacturing/doctype/production_order/production_order.py]: assign from_time and to_time to now in if condition

**Comment by nabinhait** [comment]
patch?

**Comment by jinayG** [comment]
From where do we track operations?

**Pull request merged**